pkg/services/internal/whatsonchain/broadcast.go

```diff
@@ -26,60 +25,113 @@ type broadcastRequest struct {
+
+	var resp txBroadcastResponse
 
 	req := woc.httpClient.
 		R().
 		SetContext(ctx).
-		SetResult(&txResp).
+		SetResult(&resp).
 		SetBody(broadcastRequest{TxHex: rawTxHex}).
 		AddRetryCondition(retryOnTooManyRequestsStatus)
 
-	url := fmt.Sprintf("%s/tx/raw", woc.url)
 	res, err := req.Post(url)
 	if err != nil {
 		return StatusError, "", fmt.Errorf("failed to send request to WoC: %w", err)
 	}
 
-	responseText := res.String()
-
 	if res.StatusCode() != http.StatusOK {
+		responseText := res.String()
+
 		switch {
-		case strings.Contains(responseText, "already in mempool"):
+		case containsI(responseText, "already in mempool"), containsI(responseText, "txn-already-known"):
 			return StatusAlreadyBroadcasted, txid, nil
-		case strings.Contains(responseText, "txn-mempool-conflict"):
+		case containsI(responseText, "txn-mempool-conflict"):
 			return StatusDoubleSpend, txid, nil
-		case strings.Contains(responseText, "missing inputs"):
+		case containsI(responseText, "missing inputs"):
 			return StatusMissingInputs, txid, nil
 		default:
-			return StatusError, "", fmt.Errorf("woc returned error %d: %s", res.StatusCode(), responseText)
+			return StatusError, "", fmt.Errorf("woc returned unexpected error %d: %s", res.StatusCode(), responseText)
 		}
 	}
 
-	if txResp.Txid != txid {
-		return StatusError, txResp.Txid, fmt.Errorf("woc returned txid %s does not match local calculated txid %s", txResp.Txid, txid)
+	if resp.TxID != txid {
+		return StatusError, "", fmt.Errorf("txid mismatch: expected %s, got %s", txid, resp.TxID)
 	}
 
 	return StatusSuccess, txid, nil
 }
 
-func convertNotes(notes []string) wdk.Notes {
-	converted := make(wdk.Notes, len(notes))
-	for i, note := range notes {
-		now := time.Now()
-		converted[i] = wdk.ReqHistoryNote{
-			When: &now,
-			What: note,
-			Args: nil,
+func (woc *WhatsOnChain) fetchTxInfo(ctx context.Context, txid string) (*txInfoResponse, error) {
+	var info txInfoResponse
+
+	req := woc.httpClient.
+		R().
+		SetContext(ctx).
+		SetResult(&info).
+		AddRetryCondition(retryOnTooManyRequestsStatus)
+
+	url := fmt.Sprintf("%s/tx/%s/info", woc.url, txid)
+	res, err := req.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch tx info: %w", err)
+	}
+
+	if res.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("failed to retrieve tx info: status %d, body: %s", res.StatusCode(), res.String())
+	}
+
+	return &info, nil
+}
+
+func (woc *WhatsOnChain) processSingleTx(ctx context.Context, txid string, rawTx []byte) wdk.PostedTxID {
+	status, returnedTxid, broadcastErr := woc.broadcast(ctx, rawTx)
+
+	if broadcastErr != nil {
+		return wdk.PostedTxID{
+			Result: wdk.PostedTxIDResultError,
+			TxID:   txid,
+			Error:  broadcastErr,
+			Notes:  convertNotes([]string{fmt.Sprintf("broadcast error: %v", broadcastErr)}),
 		}
 	}
-	return converted
+
+	resultStatus, notes := classifyBroadcastStatus(status)
+
+	blockHash, blockHeight, fetchErr := woc.tryFetchTxInfo(ctx, returnedTxid, &notes)
```

Comment:
Consider avoiding passing a pointer to notes and instead add the note on this level.  
```suggestion
	blockHash, blockHeight, fetchErr := woc.tryFetchTxInfo(ctx, returnedTxid)
	if fetchErr != nil {
		notes = append(notes, fetchErr.Error())
	}
```